### PR TITLE
Remove `TheoryState::options()`

### PR DESCRIPTION
**Pull request by 4tXJ7f**
This commit removes `TheoryState::options()` by changing more classes to
`EnvObj`s.

### CODE REVIEW

File: src/theory/quantifiers/quantifiers_modules.cpp [hunk 1]
```diff
@@ -24,8 +24,9 @@ namespace cvc5 {
 namespace theory {
 namespace quantifiers {
 
-QuantifiersModules::QuantifiersModules()
-    : d_rel_dom(nullptr),
+QuantifiersModules::QuantifiersModules(Env& env)
```
Comment (ajreynol) [inline src/theory/quantifiers/quantifiers_modules.cpp:27]: This class is purely for initialization of the quantifiers strategy, it doesnt have to be an EnvObj (in another PR, I've added Env& to the initialize function).

File: src/theory/quantifiers/sygus/synth_conjecture.cpp [hunk 2]
```diff
@@ -45,18 +45,20 @@ namespace cvc5 {
 namespace theory {
 namespace quantifiers {
 
-SynthConjecture::SynthConjecture(QuantifiersState& qs,
+SynthConjecture::SynthConjecture(Env& env,
```
Comment (ajreynol) [inline src/theory/quantifiers/sygus/synth_conjecture.cpp:48]: FYI this is also converted to EnvObj in https://github.com/cvc5/cvc5/pull/7145.

**Review by ajreynol — CHANGES_REQUESTED** [review]
There is one class that doesn't have to be EnvObj.

Otherwise LGTM, note that some of these classes were converted in https://github.com/cvc5/cvc5/pull/7145; I don't think anything will be hard to resolve (depending on which PR goes in first).

File: src/theory/quantifiers/quantifiers_modules.cpp (hunk 1 above)
Comment (4tXJ7f) [reply src/theory/quantifiers/quantifiers_modules.cpp]: Makes sense. I've updated this.

**Review by ajreynol — APPROVED** [review]

**Comment by ajreynol** [comment]
Will need to resolve conflicts, else LGTM

**Pull request merged**